avm/src/vm/memory.rs

```diff
@@ -260,9 +355,20 @@ impl HandlerMemory {
   }
 
   /// Stores a nested fractal of data in a given address.
-  pub fn write_fractal(self: &mut HandlerMemory, addr: i64, val: &FractalMemory) {
+  pub fn write_fractal(self: &mut HandlerMemory, addr: i64, fractal: &FractalMemory) {
     let a = self.mems.len();
-    self.mems.push(val.block.clone());
+    if !fractal.belongs(self) {
+      if fractal.hm_addr.is_none() {
+        panic!("Writing a forked/read-only FractalMemory that is also deeply-nested is not possible");
+      }
```

Comment:
I feel like this one is gonna bite us at some point...

We can delay it for now, but I suspect that we'll need to clone the `Arc` here, `transfer` the memory, and then do the write.

Reply:
Yes, but I didn't want to write the entire logic since it would be untested and this PR took long enough so I left a good error message for our future selves :)